src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2660,7 +2662,19 @@ MultiRouterPlannableQuery(Query *query, RelationRestrictionContext *restrictionC
 			char partitionMethod = PartitionMethod(distributedTableId);
 
 			if (!(partitionMethod == DISTRIBUTE_BY_HASH || partitionMethod ==
-				  DISTRIBUTE_BY_NONE))
+				  DISTRIBUTE_BY_NONE || partitionMethod == DISTRIBUTE_BY_RANGE))
+			{
+				return false;
+			}
+
+			hasHashDistributedTable |= partitionMethod == DISTRIBUTE_BY_HASH;
+			hasRangeDistributedTable |= partitionMethod == DISTRIBUTE_BY_RANGE;
+
+			/*
+			 * we could probably relax this, however, we should not
```

Comment:
(I agree your comment that we should not promote this, but, I'm more inclined to allow it.)